pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -633,31 +635,144 @@ func clusterOperatorIsNotProgressingWhenMachineConfigIs(events monitorapi.Interv
 	duration := stop.Sub(start).Seconds()
 
 	eventsByOperator := getEventsByOperator(eventsInUpgradeWindows)
-	for _, mcEvent := range eventsByOperator["machine-config"] {
-		condition := monitorapi.GetOperatorConditionStatus(mcEvent)
-		if condition == nil {
-			continue // ignore non-condition intervals
+	coProgressingStart := map[string]time.Time{}
+	for _, operatorName := range platformidentification.KnownOperators.List() {
+		for _, mcEvent := range eventsByOperator[operatorName] {
+			condition := monitorapi.GetOperatorConditionStatus(mcEvent)
+			if condition == nil {
+				continue // ignore non-condition intervals
+			}
+			if condition.Type == configv1.OperatorProgressing && condition.Status == configv1.ConditionTrue {
+				coProgressingStart[operatorName] = mcEvent.To
+				if operatorName == "machine-config" {
+					machineConfigProgressingStart = mcEvent.To
+				}
+				break
+			}
 		}
-		if condition.Type == configv1.OperatorProgressing && condition.Status == configv1.ConditionTrue {
-			machineConfigProgressingStart = mcEvent.To
-			break
+	}
+
+	except := func(co string, _ string) string {
+		intervals, ok := COWaiting[co]
+		if !ok {
+			// CO have not shown up in CVO Progressing message
+			return fmt.Sprintf("%s completing its update so fast that CVO did not recogize any waiting", co)
+		}
+		from, to := fromAndTo(intervals)
+		if d := to.Sub(from); d < 2*time.Minute {
+			// CO showed up in CVO Progressing message but the total duration is less than two minutes
+			return fmt.Sprintf("%s completing its update within less than two minutes: %s", co, d.String())
+		}
+		switch co {
+		case "cloud-controller-manager":
+			return "https://issues.redhat.com/browse/OCPBUGS-64852"
 		}
+		return ""
 	}
 
-	mcTestCase := &junitapi.JUnitTestCase{
-		Name:     fmt.Sprintf("[bz-Machine Config Operator] clusteroperator/machine-config must go Progressing=True during an upgrade test"),
-		Duration: duration,
+	// Each cluster operator must report Progressing=True during cluster upgrade
+	for _, operatorName := range platformidentification.KnownOperators.List() {
+		bzComponent := platformidentification.GetBugzillaComponentForOperator(operatorName)
+		name := fmt.Sprintf("[bz-%s] clusteroperator/%s must go Progressing=True during an upgrade test", bzComponent, operatorName)
```

Comment:
the list of known operators is presumably reasonably stable, but if any of them were to be removed or renamed across releases, the corresponding test mappings would need updating per https://github.com/openshift-eng/ci-test-mapping?tab=readme-ov-file#renaming-tests